src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -93,12 +122,48 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
     }
 
     async deinitialize(): Promise<InternalCheckoutSelectors> {
+        this._deinitializePollingTimer();
         this._orderId = undefined;
         this._paypalCommercePaymentProcessor.deinitialize();
 
         return Promise.resolve(this._store.getState());
     }
 
+    private _initializePollingMechanism(submitForm: () => void, gatewayId?: string) {
+        if (gatewayId === PaymentStrategyType.PAYPAL_COMMERCE_ALTERNATIVE_METHODS) {
+            this._pollingInterval = window.setTimeout(async () => {
+                try {
+                    if (this._pollingTimer !== undefined) {
```

Comment:
This check is redundant because `_pollingTimer` will always be a `number` (`private _pollingTimer: number = 0`). 

Also, could you please check all same conditions?

Reply:
fixed